tests/serverpod_test_server/test_integration/serverpod_cli_args_override_test.dart

```diff
@@ -0,0 +1,342 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_shared/serverpod_shared.dart';
+import 'package:serverpod_test_server/src/generated/endpoints.dart';
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+void main() {
+  // These tests verify that command line arguments correctly override
+  // ServerpodConfig values when both are provided to the Serverpod constructor.
+  // Note: runMode has special handling where it's calculated from CLI args first,
+  // then environment variables, then defaults to 'development'.
+  group(
+      'Given a ServerpodConfig with specific values and command line args that override them',
+      () {
+    test(
+        'when runMode is set to development in config and production in CLI args '
+        'then server uses production mode', () {
+      final config = ServerpodConfig(
+        runMode: 'development',
+        serverId: 'test-server',
+        role: ServerpodRole.monolith,
+        loggingMode: ServerpodLoggingMode.normal,
+        applyMigrations: false,
+        applyRepairMigration: false,
+        apiServer: ServerConfig(
+          port: 8080,
+          publicHost: 'localhost',
+          publicPort: 8080,
+          publicScheme: 'http',
+        ),
+        maxRequestSize: 524288,
+      );
+
+      final serverpod = Serverpod(
+        ['--mode', 'production'],
+        Protocol(),
+        Endpoints(),
+        config: config,
+      );
+
+      expect(serverpod.config.runMode, equals('production'));
+      // Other values should remain from config
+      expect(serverpod.config.serverId, equals('test-server'));
+      expect(serverpod.config.role, equals(ServerpodRole.monolith));
+      expect(serverpod.config.loggingMode, equals(ServerpodLoggingMode.normal));
+      expect(serverpod.config.applyMigrations, isFalse);
+      expect(serverpod.config.applyRepairMigration, isFalse);
+    });
```

Comment:
Suggestion: Do not validate other fields in a test that is for testing the production mode. They are an indirect validation and has nothing to do with the check we are validating now. If we split them we validate their rules in isolation, meaning we do not have to look at this test or modify this test if we want to change the rules for an unrelated field.

Each command line variable should have a positive and a negative test. E.g. what happens when we supply a command line argument for that specific variable and what happens when we don't